### DISCUSSION TITLE
How to deal with non React Component request?

### DISCUSSION THREAD

**Discussion by himself65** [wakujs/waku#290] [Q&A]
- JSONResponse
- StringStreamingResponse
- `rss.xml` request

In next.js, they have `/api` folder for non-react component req/res. Not sure how waku deals with this

**Comment by dai-shi** [comment]
We don't know the future, but for now the Waku architecture is primarily to provide a middleware. `cli.ts` doesn't cover all use cases. (For example, it doesn't do history api fallback for SPA, which is considerable to support.) The middleware is a building block so that you can use in your server. `08_cookies` example is one case.

As Waku becomes to more form a framework, we may support more capabilities in `cli.ts`. But, that's the current solution.

**Reply by himself65** [reply]
I think it's better to tell people how to customize the entry js for vercel, deno and cloudflare worker?

**Reply by dai-shi** [reply]
yeah, that'd be necessary.
